pages/sdk-protocol-kit/guides/safe-signatures.md

```diff
@@ -0,0 +1,844 @@
+
+```typescript
+safeTx = await protocolKit.signTransaction(safeTx, SigningMethod.ETH_SIGN); // EIP-191 signature from owner1
+protocolKit = await protocolKit.connect({ ethAdapter: ethAdapter2 }); // Connect another owner
+safeTx = await protocolKit.signTransaction(
+  safeTx,
+  SigningMethod.ETH_SIGN_TYPED_DATA_V4
+); // owner2 EIP-712 typed data signature
+```
+
+In this snippet, we add the signature for `owner1`. Then, we use the `connect()` method to connect the `owner2` adapter and sign again to add the second signature.
+
+If we examine the `safeTx` object at this point, we should see something similar to the following:
+
+```javascript
+EthSafeTransaction {
+  signatures: Map(2) {
+    '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1' => EthSafeSignature {
+      signer: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1',
+      data: '0x969308e2abeda61a0c9c41b3c615012f50dd7456ca76ea39a18e3b975abeb67f275b07810dd59fc928f3f9103e52557c1578c7c5c171ffc983afa5306466b1261f',
+      isContractSignature: false
+    },
+    '0xffcf8fdee72ac11b5c542428b35eef5769c409f0' => EthSafeSignature {
+      signer: '0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0',
+      data: '0x4d63c79cf9d743782bc31ad58c1a316020b39839ab164caee7ecac9829f685cc44ec0d066a5dfe646b2ffeeb37575df131daf9c96ced41b8c7c4aea8dc5461801c',
+      isContractSignature: false
+    }
+  },
+  data: { ... }
+}
+```
+
+The `data` prop in the `signatures` map represents the concrete signature. The `isContractSignature` flag (_false_) indicates if the signature is an Ethereum signature or a Contract signature (A signer Safe Account).
+
+An Ethereum signature is composed of two 32-byte integers (r, s) and an extra byte for recovery id (v), making a total of 65 bytes. In hexadecimal string format, each byte is represented by 2 characters. Hence, a 65-byte Ethereum signature will be 130 characters long. Including the '0x' prefix commonly used with signatures, the total character count for such a signature would be 132. For a more detailed explanation, you can refer to [this link](https://docs.safe.global/safe-smart-account/signatures) for more information.
+
+> To represent a byte (8 bits) in hexadecimal, you need 2 characters. Each hexadecimal character represents 4 bits. Therefore, 2 hexadecimal characters (2 x 4 bits) are able to represent a byte (8 bits).
+
+The final part of the signature, either `1f` or `1c`, indicates the signature type.
+
+The hexadecimal number `1f` converts to the decimal number 31. It indicates that the signature is an `eth_sign` because the number is greater than 30. This adjustment is made for the `eth_sign` flow in the contracts. You can find the relevant code [here](https://github.com/safe-global/safe-contracts/blob/f03dfae65fd1d085224b00a10755c509a4eaacfe/contracts/Safe.sol#L344-L347).
```

Comment:
```suggestion
The hexadecimal value `1f` equals to the decimal number 31. Because the decimal value is greater than 30,  it [indicates to the Safe smart contract]()(https://github.com/safe-global/safe-smart-account/blob/f03dfae65fd1d085224b00a10755c509a4eaacfe/contracts/Safe.sol#L344-L347) that the signature is an `eth_sign`.
```